src/deepsparse/server/openai_server.py

```diff
@@ -93,49 +93,26 @@ async def create_chat_completion(raw_request: Request):
             request = ChatCompletionRequest(**await raw_request.json())
             _LOGGER.debug("Received chat completion request %s" % request)
 
-            if isinstance(request.messages, str):
-                prompt = request.messages
-            else:
-                # else case assumes a FastChat-compliant dictionary
-                # Fetch a model-specific template from FastChat
-                try:
-                    from fastchat.model.model_adapter import get_conversation_template
-                except ImportError as e:
-                    return create_error_response(
-                        HTTPStatus.FAILED_DEPENDENCY,
-                        f"{str(e)} - Please ensure `fastchat` is installed.",
-                    )
-
-                conv = get_conversation_template(request.model)
-                messages = request.messages
-                messages = messages if isinstance(messages, list) else [messages]
-                # add the model to the Conversation template, based on the given role
-                for message in messages:
-                    msg_role = message["role"]
-                    if msg_role == "system":
-                        conv.system_message = message["content"]
-                    elif msg_role == "user":
-                        conv.append_message(conv.roles[0], message["content"])
-                    elif msg_role == "assistant":
-                        conv.append_message(conv.roles[1], message["content"])
-                    else:
-                        return create_error_response(
-                            HTTPStatus.BAD_REQUEST, "Message role not recognized"
-                        )
-
-                # blank message to start generation
-                conv.append_message(conv.roles[1], None)
-                prompt = conv.get_prompt()
-
-            request_id = f"cmpl-{random_uuid()}"
-            created_time = int(time.time())
             model = request.model
-
             pipeline = app.model_to_pipeline.get(model)
             if not pipeline:
                 return create_error_response(
-                    HTTPStatus.BAD_REQUEST, f"{model} is not available"
+                    HTTPStatus.BAD_REQUEST,
+                    f"The model `{model}` does not exist.",
+                )
+
+            try:
+                prompt = pipeline.tokenizer.apply_chat_template(
+                    conversation=request.messages,
```

Comment:
This works for both a `dictionary` and a `string`? any difference in how they're both treated?

Reply:
good catch, it definitely does not deal with `str` properly, so vLLM has a bug there :)